tools/autograd/gen_python_functions.py

```diff
@@ -290,8 +300,26 @@ def parse_arg(arg, unpack_args=False):
             env['dispatch_call'] = 'default_type().{}'.format(declaration['name'])
         env['AutoNoGIL'] = 'AutoNoGIL no_gil;'
         env['AutoGPU'] = auto_gpu(declaration)
+
+        requires_grad = None
+        if len(declaration.get('python_binding_arguments', [])) > 0:
```

Comment:
This outer if looks unnecessary?

Reply:
It's not as written because I'm lazy later about using declaration['python_binding_arguments'], but I'll just use get consistently.